src/fiatconnect/kyc/KycDenied.tsx

```diff
@@ -1,27 +1,139 @@
+import fontStyles from 'src/styles/fonts'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+
+import { useSelector } from 'react-redux'
+import { localCurrencyExchangeRatesSelector } from 'src/localCurrency/selectors'
+import getNavigationOptions from 'src/fiatconnect/kyc/getNavigationOptions'
 
 type Props = StackScreenProps<StackParamList, Screens.KycDenied>
 
-// TODO implement designs:
-// (retryable): https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=2887%3A41469
-// (final attempt): https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=2885%3A41683
-function KycDenied(_props: Props) {
-  return (
-    <SafeAreaView style={styles.container}>
-      <Text>KYC Denied</Text>
-    </SafeAreaView>
-  )
+function KycDenied({ route, navigation }: Props) {
+  const { quote, flow, retryable } = route.params
+
+  const exchangeRates = useSelector(localCurrencyExchangeRatesSelector)
+  useMemo(() => {
+    navigation.setOptions(
+      getNavigationOptions({
+        fiatConnectKycStatus: FiatConnectKycStatus.KycDenied,
+        quote,
+        exchangeRates,
+      })
+    )
+  }, [exchangeRates])
+
+  const { t } = useTranslation()
+
+  const onPressTryAgain = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_kyc_status_try_again, {
+      provider: quote.getProviderId(),
+      flow,
+      fiatConnectKycStatus: FiatConnectKycStatus.KycDenied,
+    })
+    navigate(Screens.KycLanding, {
+      quote,
+      flow,
+      step: 'one',
+    })
+  }
+  const onPressSwitch = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_kyc_status_switch_method, {
+      provider: quote.getProviderId(),
+      flow,
+      fiatConnectKycStatus: FiatConnectKycStatus.KycDenied,
+    })
+    navigate(Screens.SelectProvider, {
+      flow,
+      selectedCrypto: quote.getCryptoType(),
+      amount: {
+        crypto: Number(quote.getCryptoAmount()),
+        fiat: Number(
+          convertCurrencyToLocalAmount(
+            quote.getCryptoAmount(),
+            exchangeRates[quote.getCryptoType()]
+          )
+        ),
+      },
+    })
+  }
+  if (retryable) {
+    return (
+      <SafeAreaView style={styles.container}>
+        <Text style={styles.title}>{t('fiatConnectKycStatusScreen.denied.retryable.title')}</Text>
+        <Text testID="descriptionText" style={styles.description}>
+          {t('fiatConnectKycStatusScreen.denied.retryable.description')}
+        </Text>
+        <Button
+          style={styles.tryAgainButton}
+          testID="tryAgainButton"
+          onPress={onPressTryAgain}
+          text={t('fiatConnectKycStatusScreen.denied.retryable.tryAgain')}
+          type={BtnTypes.PRIMARY}
+          size={BtnSizes.MEDIUM}
+        />
+        <Button
+          style={styles.switchButton}
+          testID="switchButton"
+          onPress={onPressSwitch}
+          text={t('fiatConnectKycStatusScreen.expired.switch')}
```

Comment:
did you mean to use the copy for "expired" here?

Reply:
from the naming it sounds like it's just for expired

Reply:
good catch, updated.